spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -100,142 +66,194 @@
     end
 
     it "picks correct base image for Lantern" do
-      expect(PostgresResource).to receive(:[]).and_return(postgres_resource)
-      expect(postgres_resource).to receive(:flavor).and_return(PostgresResource::Flavor::LANTERN).at_least(:once)
-      expect(Prog::Vm::Nexus).to receive(:assemble_with_sshable).with(anything, hash_including(boot_image: "postgres16-lantern-ubuntu-2204")).and_return(instance_double(Strand, id: "62c62ddb-5b5a-4e9e-b534-e73c16f86bcb", subject: instance_double(Vm, add_vm_firewall: nil)))
-      expect(PostgresServer).to receive(:create_with_id).and_return(instance_double(PostgresServer, id: "5c13fd6a-25c2-4fa4-be48-2846f127526a"))
-      described_class.assemble(resource_id: postgres_resource.id, timeline_id: "91588cda-7122-4d6a-b01c-f33c30cb17d8", timeline_access: "push", representative_at: Time.now)
+      postgres_project = Project.create(name: "default")
+      expect(Config).to receive(:postgres_service_project_id).and_return(postgres_project.id).at_least(:once)
+
+      # Create a lantern-flavored postgres resource
+      lantern_resource = PostgresResource.create(
+        project_id: user_project.id,
+        location_id: Location::HETZNER_FSN1_ID,
+        name: "pg-lantern",
+        target_vm_size: "standard-2",
+        target_storage_size_gib: 64,
+        superuser_password: "dummy-password",
+        target_version: "16",
+        flavor: PostgresResource::Flavor::LANTERN
+      )
+      # Create firewall with correct name after resource exists
+      Firewall.create(name: "#{lantern_resource.ubid}-internal-firewall", location_id: Location::HETZNER_FSN1_ID, project_id: postgres_project.id)
+      postgres_timeline = PostgresTimeline.create
+
+      st = described_class.assemble(resource_id: lantern_resource.id, timeline_id: postgres_timeline.id, timeline_access: "push", representative_at: Time.now)
+      postgres_server = PostgresServer[st.id]
+      expect(postgres_server.vm.boot_image).to eq("postgres16-lantern-ubuntu-2204")
     end
 
     it "picks correct base image for AWS-pg16" do
-      expect(PostgresResource).to receive(:[]).and_return(postgres_resource)
-      loc = Location.create(
-        name: "us-west-2",
-        display_name: "aws-us-west-2",
-        ui_name: "aws-us-west-2",
-        visible: true,
-        provider: "aws",
-        project_id: user_project.id
+      # Use an existing seed location (us-west-2) that has billing rates
+      aws_location = Location.find(name: "us-west-2")
+      skip "us-west-2 location not available in test seeds" unless aws_location
+
+      postgres_project = Project.create(name: "default")
+      expect(Config).to receive(:postgres_service_project_id).and_return(postgres_project.id).at_least(:once)
+
+      # Ensure AMI mapping exists
+      PgAwsAmi.find_or_create(aws_location_name: "us-west-2", pg_version: "16", arch: "x64") { |a| a.aws_ami_id = "ami-pg16-x64" }
```

Comment:
It's odd to use `find_or_create` inside a spec. You should know at this point whether or not the `PgAwsAmi` should exist, and use either `first` or `create`.